### PR TITLE
fix l2coinbase

### PR DESCRIPTION
**Pull request by BananaLF**
Closes #<issue number>.

### Reviewers

Main reviewers:

- @John
- @Doe

Codeowner reviewers:

- @-Alice
- @-Bob

### CODE REVIEW

File: sequencesender/sequencesender.go [hunk 1]
```diff
@@ -186,11 +190,18 @@ func (s *SequenceSender) getSequencesToSend(ctx context.Context) ([]types.Sequen
...
 			if err != nil {
-				return nil, err
+				return nil, l2coinbase, err
 			}
 			seq.ForcedBatchTimestamp = forcedBatch.ForcedAt.Unix()
 		}
 
+		//  All coinbase of sequences must be same
+		if len(sequences) == 0 {
+			l2coinbase.SetBytes(batch.Coinbase.Bytes())
```
Comment (zjg555543) [inline sequencesender/sequencesender.go:200]: Could it can be add ut to cover this case?

File: config/default.go [hunk 2]
```diff
@@ -116,8 +117,8 @@ MaxTxLifetime = "3h"
 [SequenceSender]
 WaitPeriodSendSequence = "5s"
 LastBatchVirtualizationTimeMaxWaitPeriod = "5s"
+MaxTxSizeForL1 = 10
```
Comment (zjg555543) [inline config/default.go:120]: May be it's 131072

File: sequencesender/test.keystore [hunk 3]
```diff
@@ -0,0 +1 @@
+{"address":"f39fd6e51aad88f6f4ce6ab8827279cfffb92266","crypto":{"cipher":"aes-128-ctr","ciphertext":"d005030a7684f3adad2447cbb27f63039eec2224c451eaa445de0d90502b9f3d","cipherparams":{"iv":"dc07a54bc7e388efa89c34d42f2ebdb4"},"kdf":"scrypt","kdfparams":{"dklen":32,"n":262144,"p":1,"r":8,"salt":"cf2ec55ecae11171de575112cfb16963570533a9c46fb774473ceb11519eb24a"},"mac":"3eb180d405a5da6e462b2adc00091c14856c91d574bf27348714506357d6e177"},"id":"035454db-6b6d-477f-8a79-ce24c10b185f","version":3}
```
Comment (zjg555543) [inline sequencesender/test.keystore:1]: Can it use file test/sequencer.keystore ?

File: sequencesender/sequencesender.go [hunk 4]
```diff
@@ -85,7 +87,7 @@ func (s *SequenceSender) tryToSendSequence(ctx context.Context, ticker *time.Tic
 
 	// Check if should send sequence to L1
 	log.Infof("getting sequences to send")
-	sequences, err := s.getSequencesToSend(ctx)
+	sequences, l2coinbase, err := s.getSequencesToSend(ctx)
 	if err != nil || len(sequences) == 0 {
```
Comment (zjg555543) [inline sequencesender/sequencesender.go:91]: check l2coinbase is nill

File: sequencer/finalizer.go [hunk 5]
```diff
@@ -962,11 +963,29 @@ func (f *finalizer) syncWithState(ctx context.Context, lastBatchNum *uint64) err
...
-		if err != nil {
-			return fmt.Errorf("failed to get work-in-progress batch, err: %w", err)
+		if bytes.Equal(lastBatch.Coinbase.Bytes(), f.l2coinbase.Bytes()) {
+			f.batch, err = f.dbManager.GetWIPBatch(ctx)
+			if err != nil {
+				return fmt.Errorf("failed to get work-in-progress batch, err: %w", err)
+			}
+		} else {
+			oldGER := lastBatch.GlobalExitRoot
+			if err := f.dbManager.DeleteBatchByNumber(ctx, *lastBatchNum, nil); err != nil {
```
[the remark this answers is not in the document]
Comment (BananaLF) [reply sequencer/finalizer.go:973]: it can not cause problem

File: sequencer/interfaces.go [hunk 6]
```diff
@@ -100,7 +101,7 @@ type workerInterface interface {
 type dbManagerInterface interface {
 	OpenBatch(ctx context.Context, processingContext state.ProcessingContext, dbTx pgx.Tx) error
 	BeginStateTransaction(ctx context.Context) (pgx.Tx, error)
-	CreateFirstBatch(ctx context.Context, sequencerAddress common.Address) state.ProcessingContext
+	CreateFirstBatch(ctx context.Context, l2coinbase common.Address) state.ProcessingContext
```
[the remark this answers is not in the document]
Comment (BananaLF) [reply sequencer/interfaces.go:104]: only genesis start can call CreateFistBatch

**Review by zjg555543 — APPROVED** [review]

**Pull request closed without merging**